### PR TITLE
[BIOMAGE-1990] - Add notification on deploy failure

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
Send Slack notification to the #pipelines channel if there is a failure in deployment.

The notification is added to the following scripts:
- Deploy changed cf
- Deploy infra

2 other scripts which I think need the notification are these scripts:
- Delete old staging step functions
- Backup Cognito users

They scripts already have the notification implemented

#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1990

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock  (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/hms-dbmi-cellenics/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: .github/workflows/deploy-changed-cf.yaml [hunk 1]
```diff
@@ -306,4 +306,20 @@ jobs:
...
+    name: Report if workflow failed
+    runs-on: ubuntu-20.04
+    needs: [setup, get-modified-templates, lint-templates, deploy-templates, setup-rds-roles]
+    if: failure()
+    steps:
+      - id: send-to-slack
+        name: Send failure notification to Slack on failure
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.BUILD_STATUS_BOT_TOKEN }}
+        uses: voxmedia/github-action-slack-notify-build@v1
```
Comment (ivababukova) [inline .github/workflows/deploy-changed-cf.yaml:321]: have you configured this to post the failure only if the error is on the master branch? We don't care about failures in custom branches. Or do you think that we do?

File: .github/workflows/deploy-infra.yaml [hunk 2]
```diff
@@ -497,3 +497,19 @@ jobs:
...
+
+  report-if-failed:
+    name: Report if workflow failed
+    runs-on: ubuntu-20.04
+    needs: [setup, create-eks-cluster, configure-cluster]
+    if: failure()
+    steps:
+      - id: send-to-slack
+        name: Send failure notification to Slack on failure
+        env:
```
Comment (ivababukova) [inline .github/workflows/deploy-infra.yaml]: same here -- we want to see failures if they are with the master branch. If they are not with the master branch, most likely  a developer is testing and then the notification will just be noise

File: .github/workflows/deploy-changed-cf.yaml (hunk 1 above)
Comment (aerlaut) [reply .github/workflows/deploy-changed-cf.yaml:321]: Hmm.. yea, that's right. If a developer is testing, they'll pay attention whether the. branch is merged successfully or not. I'll add that check in.

File: .github/workflows/deploy-infra.yaml (hunk 2 above)
Comment (aerlaut) [reply .github/workflows/deploy-infra.yaml]: Same as above. If a developer is testing, they'll pay attention to the run.

**Review by ivababukova — APPROVED** [review]

**Pull request merged**